x-pack/legacy/plugins/apm/public/components/app/Settings/CustomizeUI/CustomLink/__test__/helper.test.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import {
+  convertFiltersToArray,
+  convertFiltersToObject,
+  getSelectOptions
+} from '../CustomLinkFlyout/helper';
+import { CustomLink } from '../../../../../../../../../../plugins/apm/server/lib/settings/custom_link/custom_link_types';
+
+describe('Custom link helper', () => {
+  describe('convertFiltersToArray', () => {
+    it('returns array of tuple when custom link not defined', () => {
+      expect(convertFiltersToArray()).toEqual([['', '']]);
+    });
+    it('returns filters as array', () => {
+      expect(
+        convertFiltersToArray({
+          'service.name': 'foo',
+          'transaction.type': 'bar'
+        } as CustomLink)
+      ).toEqual([
+        ['service.name', 'foo'],
+        ['transaction.type', 'bar']
+      ]);
+    });
+    it('returns empty when no filter is added', () => {
+      expect(
+        convertFiltersToArray({
+          label: 'foo',
+          url: 'bar'
+        } as CustomLink)
+      ).toEqual([['', '']]);
+    });
+  });
+
+  describe('convertFiltersToObject', () => {
+    it('returns undefined when any filter is added', () => {
+      expect(convertFiltersToObject([['', '']])).toBeUndefined();
+    });
+    it('removes uncompleted filters', () => {
+      expect(
+        convertFiltersToObject([
+          ['service.name', ''],
+          ['', 'foo'],
+          ['transaction.type', 'bar']
+        ])
+      ).toEqual({ 'transaction.type': ['bar'] });
+    });
+    it('splits the value by comma', () => {
+      expect(
+        convertFiltersToObject([
+          ['service.name', 'foo'],
+          ['service.environment', 'foo, bar'],
+          ['transaction.type', 'foo, '],
+          ['transaction.name', 'foo,']
+        ])
+      ).toEqual({
+        'service.name': ['foo'],
+        'service.environment': ['foo', 'bar'],
+        'transaction.type': ['foo'],
+        'transaction.name': ['foo']
+      });
+    });
+  });
+
+  describe('getSelectOptions', () => {
+    it('returns all available options when no filters were selected', () => {
+      expect(
+        getSelectOptions(
+          [
+            ['', ''],
+            ['', ''],
+            ['', ''],
+            ['', '']
+          ],
+          ''
```

Comment:
If no filters are selected, wouldn't the list of filters be empty so `getSelectOptions([])` ?

Reply:
This test should check when a user adds all filters, clicking on `Add another filter`, but he doesn't select any from the list, which in case creates four empty Tuples.